### PR TITLE
polylineencoder: add v1.1.2 + support fPIC + use CMakeToolchain

### PR DESCRIPTION
**Pull request by wolfee001**
update polyline lib with fPIC option

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/polylineencoder/all/conanfile.py [hunk 1]
```diff
@@ -1,51 +1,69 @@
...
     def package(self):
-        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
 
     def package_info(self):
-        self.cpp_info.libs.append("polylineencoder")
+        if self.version == "1.0.0":
```
Comment (uilianries) [inline recipes/polylineencoder/all/conanfile.py:64]: What about 1.1.2?

Comment (wolfee001) [reply recipes/polylineencoder/all/conanfile.py:64]: 1.1.2 is header-only. in my understanding you don't have to specify the library for header-only packages. Am i wrong?

Comment (uilianries) [reply recipes/polylineencoder/all/conanfile.py:64]: Correct, I missed the package Id part. Thank you for clarifying.

File: recipes/polylineencoder/all/conanfile.py [hunk 2]
```diff
@@ -1,51 +1,69 @@
...
+            self.cpp_info.libs.append("polylineencoder")
+
+    def package_id(self):
+        if self.version == "1.1.2":
```
Comment (uilianries) [inline recipes/polylineencoder/all/conanfile.py:68]:
```suggestion
        if Version(self.version) >= "1.1.2":
```
I presume, for any future version, this project will continue to be header-only, correct?

Comment (wolfee001) [reply recipes/polylineencoder/all/conanfile.py:68]: i have no idea, the guy just broke the interface in the very same major too... so let's hope for the best :D

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by prince-chrismc** [comment]
I woud be nice to have a more descriptive title to highlight the amazing work that was done 🙏 

`polylineencoder: add v1.1.2 + support fPIC + use CMakeToolchain` would be my suggestion

**Comment by wolfee001** [comment]
> It would be nice to have a more descriptive title to highlight the amazing work that was done 🙏
> 
> `polylineencoder: add v1.1.2 + support fPIC + use CMakeToolchain` would be my suggestion

image

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**